pkg/image/containerdregistry/registry.go

```diff
@@ -91,7 +91,7 @@ func (r *Registry) Pull(ctx context.Context, ref image.Reference) error {
 
 // Unpack writes the unpackaged content of an image to a directory.
 // If the referenced image does not exist in the registry, an error is returned.
-func (r *Registry) Unpack(ctx context.Context, ref image.Reference, dir string) error {
+func (r *Registry) Unpack(ctx context.Context, ref image.Reference, from, to string) error {
```

Comment:
I just thought of an alternative to changing the interface here: Create a new type that implements `image.Reference` but maybe also implements some other interface that we can make an assertion on; e.g.

```
type RootedReference interface {
    Reference
    RootDir() string
}

type SimpleRootedReference struct {
    SimpleReference
    Root
}

func (r SimpleRootedReference) RootDir() string {
    return r.Root
}
```

Here's a playground link that makes sure this can be used the way I think it can: https://play.golang.org/p/tlF1e_joEHo

Reply:
I think I'm following, but we end up mixing the image reference with the filepath on the system in the same type right? It's kind of confusing. 

In a previous review you mentioned this interface being underspecified, I agreed with that. I think we should change it, if we are ok with making a change to a public interface, which I believe we are.

Reply:
> It's kind of confusing.

I could see that view point. My head was automatically phrasing it as "a higher fidelity reference" type.

> In a previous review you mentioned this interface being underspecified

Yeah, I was just trying to come up with a way to work with the existing interfaces in a way that could make sense.

> I think we should change it, if we are ok with making a change to a public interface, which I believe we are.

I think the risk is low for this particular interface, although I would like to get one more set of eyes on this to see what they think. I don't think this is worth agonizing over at this point though.

Reply:
Spoke with @benluddy on this and got some positive feedback for the interface change.